### ISSUE TITLE
Swoole\Event::wait() consumes 100% cpu

### ISSUE THREAD

**Issue by miketsl** [swoole/swoole-src#3285]
Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a simple script for reproducing the error.
```php
<?php
class Worker {
        public function __construct() {
		$this->run();
        }

        public function run() {
                swoole_event_add(STDIN,
                        function ($pipe) {
                                $data=trim(fgets($pipe));
                                if($data=="exit") {
                                        die;
                                }
                                else {
					usleep(10000);
					echo "I got some data\n";
                                }
                        }
                );
		Swoole\Event::wait();
        }
}

new Worker();
?>
```

htop shows 100% cpu consumption

2. What did you expect to see?
Low or no CPU consumption .. for example if you do:
```php
<php
while(1) {
    usleep(100);
}
?>
```

You will get 0 usage or very low usage on htop.

3. What did you see instead?

100% cpu utilization

4. What version of Swoole are you using (show your `php --ri swoole`)?
php --ri swoole

swoole

Swoole => enabled
Author => Swoole Team <team@swoole.com>
Version => 4.5.0
Built => Apr 26 2020 09:59:20
coroutine => enabled
epoll => enabled
eventfd => enabled
signalfd => enabled
cpu_affinity => enabled
spinlock => enabled
rwlock => enabled
pcre => enabled
zlib => 1.2.11
mutex_timedlock => enabled
pthread_barrier => enabled
futex => enabled
async_redis => enabled

Directive => Local Value => Master Value
swoole.enable_coroutine => On => On
swoole.enable_library => On => On
swoole.enable_preemptive_scheduler => Off => Off
swoole.display_errors => On => On
swoole.use_shortname => On => On
swoole.unixsock_buffer_size => 8388608 => 8388608

5. What is your machine environment used (including version of kernel & php & gcc) ?
Linux  #67-Ubuntu SMP Mon Mar 2 07:24:29 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

PHP 7.3.17-1+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Apr 19 2020 07:44:26) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.17, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.17-1+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies

gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

**Comment by miketsl** [comment]
On further review it may not be Event::wait() doing it.  This worker pasted above is started 6 times with Swoole\Process and Process->exec() .. it seems that only ONE of the workers consumes 100% cpu, and it only happens when swoole process starts it.  (if I start 6 with popen... doesn't consume 100% cpu).  Not sure even how to debug it.

**Comment by matyhtf** [comment]
Use strace and gdb to track the process of infinite loop.

```shell
strace -p PID
gdb -p PID
```

**Comment by miketsl** [comment]
php,6257
  ├─php,6258 ./mike.php
  ├─php,6259 ./mike.php
  ├─php,6260 ./mike.php
  ├─php,6261 ./mike.php
  ├─php,6262 ./mike.php
  ├─php,6263 ./mike.php
  ├─{php},6264
  ├─{php},6265
  ├─{php},6266
  ├─{php},6267
  ├─{php},6268
  └─{php},6269

ps -ef |grep 6257
ubuntu    6257  5982 99 04:42 pts/3    00:02:01 MinMax worker
ubuntu    6258  6257  0 04:42 pts/3    00:00:00 /usr/bin/php ./mike.php
ubuntu    6259  6257  0 04:42 pts/3    00:00:00 /usr/bin/php ./mike.php
ubuntu    6260  6257  0 04:42 pts/3    00:00:00 /usr/bin/php ./mike.php
ubuntu    6261  6257  0 04:42 pts/3    00:00:00 /usr/bin/php ./mike.php
ubuntu    6262  6257  0 04:42 pts/3    00:00:00 /usr/bin/php ./mike.php
ubuntu    6263  6257  0 04:42 pts/3    00:00:00 /usr/bin/php ./mike.php

sudo strace -p 6257
strace: Process 6257 attached
strace: [ Process PID=6257 runs in x32 mode. ]
strace: [ Process PID=6257 runs in 64 bit mode. ]
futex(0x557bc40e0050, FUTEX_WAIT_PRIVATE, 2, NULL) = -1 EAGAIN (Resource temporarily unavailable)
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAIT_PRIVATE, 2, NULL) = -1 EAGAIN (Resource temporarily unavailable)
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x557bc40e0050, FUTEX_WAIT_PRIVATE, 2, NULL) = -1 EAGAIN (Resource temporarily unavailable)
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAIT_PRIVATE, 2, NULL) = -1 EAGAIN (Resource temporarily unavailable)
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAIT_PRIVATE, 2, NULL) = -1 EAGAIN (Resource temporarily unavailable)
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAIT_PRIVATE, 2, NULL) = -1 EAGAIN (Resource temporarily unavailable)
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x557bc40e0050, FUTEX_WAIT_PRIVATE, 2, NULL) = -1 EAGAIN (Resource temporarily unavailable)

using 100%

**Comment by miketsl** [comment]
UPDATE:  I found the problem.  I made a main coroutine function called after the processes started up, in which I polled kafka.  This coroutine was using a while(1) on the poll of kafka but without a sleep, and that caused the 100% cpu.  I added a co::sleep(0.001) to the loop and now its doing 2.7%.  
 
(btw,.. a coroutine client for kafka would be pretty cool, the one someone worked on and is avail on github, doesn't work)...
